provider/openstack/firewaller.go

```diff
@@ -73,6 +73,12 @@ type Firewaller interface {
 	// If the model security group doesn't exist, return a NotFound error
 	ModelIngressRules(ctx context.ProviderCallContext) (firewall.IngressRules, error)
 
+	// DeleteMachineGroup deletes the security group unique to a given machine.
+	// If we're in firewall mode 'instance' there will be only one, the security
+	// group used to configure this specific instance. In 'global' mode each
```

Comment:
Pretty much the first case. The one caveat is that the sec group is created when the first instance is created. The reason for this is it was decided we didn't want to create resources until they're needed. The sec group is then destroyed when the model is destroyed

The same is true for all the clouds that support security group based firewalls, but I'll add a comment here in the openstack provider since it's important for it's usage

Reply:
Ok, so it's bounded to the model, with some lazy-loading. That's consistent.

tx

Reply:
Done